### DISCUSSION TITLE
Show Checked Items in Dropdown

### DISCUSSION THREAD

**Discussion by elliewalkersm** [nss-evening-cohort-14/evening-client-side#199] [Help Me!] [answered]
Your Name: Ellie Walker

Have you searched the other help tickets?
Yes I have.

Describe your issue
In my edit menu item form, I want the previously checked ingredients to populate and show the current ingredients assigned to the menu item.

List any error messages that appear
No error message.

What have you googled?
https://www.javatpoint.com/how-to-get-all-checked-checkbox-value-in-javascript
https://www.techiedelight.com/retrieve-checked-checkboxes-values-javascript/

What have you tried? Tell us everything you have tried.
- adding ${menuObject.ingredient} to the select-ingredient div in the edit form
- replace the checkbox empty array brackets with a document.QuerySelector("#select-ingredient").value
- replace the checkbox empty array brackets with a document.QuerySelector("#select-ingredient").checked

Links to my repo

edit form: https://github.com/nss-evening-cohort-14/michelin-star-restaurant-dashboard-michelin-star-restaurant-dashboard-1/blob/ejw-update-menu/src/javascripts/components/forms/editMenuItems.js

domevent: https://github.com/nss-evening-cohort-14/michelin-star-restaurant-dashboard-michelin-star-restaurant-dashboard-1/blob/ejw-update-menu/src/javascripts/events/domEvents.js

**Comment by elliewalkersm** [comment, marked as answer]
- passed an array = [ ] in the selectIngredient function
- added an if else statement in the selectIngredient function to look for checked or not checked
- in editMenuItemForm, I mapped through the ingredients array and return an array of firebaseKeys
- I passed through the .map in my selectIngredients function on the edit form.

**Reply by drteresavasquez** [reply]
Was this the answer? Looks very cool to me!
